### PR TITLE
Remove custom config creation in test in favor of using mocks

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Stop using custom created configuration for tests in favor of the official config mocks.

### Describe how to test/QA your changes

Running the CI is enough since only the test are modified.

### CODE REVIEW

**Review by tobz — APPROVED** [review]

File: pkg/util/http/go.mod
Comment (pgimalac) [inline pkg/util/http/go.mod:1]: **❓ question:** is it expected that there are so many new dependencies ? Are all those dependencies of the new config ?‏

**Review by mellon85 — APPROVED** [review]

File: comp/core/log/mock/go.mod [hunk 1]
```diff
@@ -42,7 +45,7 @@ require (
 	github.com/mitchellh/mapstructure v1.1.2 // indirect
 	github.com/mohae/deepcopy v0.0.0-20170929034955-c48cc78d4826 // indirect
 	github.com/pelletier/go-toml v1.2.0 // indirect
-	github.com/spf13/afero v1.1.2 // indirect
+	github.com/spf13/afero v1.11.0 // indirect
```
Comment (vickenty) [inline comp/core/log/mock/go.mod:48]: This and a few other packages had a large version jump. Is it expected? It may be better to separate version bumps in a separate PR if this is necessary.

**Review by dineshg13 — APPROVED** [review]

File: comp/core/log/mock/go.mod (hunk 1 above)
Comment (pgimalac) [reply comp/core/log/mock/go.mod:48]: This is already the current version in the main `go.mod` file so I think there is no actual change when building the agent (often other `go.mod` files are not updated...)

File: pkg/util/http/go.mod
Comment (hush-hush) [reply pkg/util/http/go.mod:1]: Not really, I rebased again and it's better now

File: comp/forwarder/defaultforwarder/go.mod [hunk 2]
```diff
@@ -56,19 +56,19 @@ require (
 	github.com/DataDog/datadog-agent/comp/core/log/mock v0.0.0-00010101000000-000000000000
 	github.com/DataDog/datadog-agent/comp/core/status v0.56.0-rc.3
 	github.com/DataDog/datadog-agent/pkg/config/mock v0.58.0-devel
-	github.com/DataDog/datadog-agent/pkg/config/model v0.56.0-rc.3
-	github.com/DataDog/datadog-agent/pkg/config/setup v0.56.0-rc.3
+	github.com/DataDog/datadog-agent/pkg/config/model v0.57.1
```
Comment (pgimalac) [inline comp/forwarder/defaultforwarder/go.mod:59]: Is it expected that it updates the version of some modules to v0.57.1 ?

**Review by pgimalac — APPROVED** [review]
Seems kind of weird that `pkg/util/http` has so many dependencies now honestly, but since they are just to create a config mock in tests that should be fine

File: comp/forwarder/defaultforwarder/go.mod (hunk 2 above)
Comment (hush-hush) [reply comp/forwarder/defaultforwarder/go.mod:59]: This is done automatically by `go mod tidy`.
